### PR TITLE
Task 1 completed

### PR DESCRIPTION
**Pull request by kely-rufino**
### Display the film collection

Display a list to browse with each film title, the year it was made, the price, whether it's in stock, and each of its genres

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,58 +1,70 @@
-Secondhand Movies
```
Comment (luciavelasco) [inline README.md:1]: Did you mean to remove this?

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+
+import AllMeetupsPage from './pages/AllMeetups';
```
Comment (luciavelasco) [inline src/App.js]: Why is this called "Meetups"? This should be DVDs

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+function App() {
+  return (
+    <div>
+      <BrowserRouter>
+      
+        <Routes>
+          <Route path='/' element={<AllMeetupsPage/>}></Route>
```
Comment (luciavelasco) [inline src/App.js]: Are you planning on adding more routes to this app?

File: src/components/meetups/MeetupItem.module.css [hunk 4]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (luciavelasco) [inline src/components/meetups/MeetupItem.module.css]: Why is this file called `*.module.css`?

File: src/components/meetups/MeetupItem.module.css [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    text-align: center;
+  }
+  
+  .actions button {
+    font: inherit;
+    cursor: pointer;
+    color: #77002e;
+    border: 1px solid #77002e;
+    background-color: transparent;
+    padding: 0.5rem 1.5rem;
```
Comment (luciavelasco) [inline src/components/meetups/MeetupItem.module.css]: I like that you're using rem here, I think it's a bit more reliable than pixels.

File: src/components/meetups/MeetupList.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import MeetupItem from './Meetupitem';
+import classes from './MeetupList.module.css';
+
+function MeetupList(props){
```
Comment (luciavelasco) [inline src/components/meetups/MeetupList.js]: Rename this

File: src/components/meetups/MeetupList.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    return (   
+        <ul className={classes.list}>
```
Comment (luciavelasco) [inline src/components/meetups/MeetupList.js]: What does this do?

File: src/components/meetups/MeetupList.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+            {props.meetups.map(meetup => ( <MeetupItem
+                key={meetup.id}
```
Comment (luciavelasco) [inline src/components/meetups/MeetupList.js]: Really good use of id as a key!!!

File: src/components/meetups/MeetupList.module.css [hunk 9]
```diff
@@ -0,0 +1,19 @@
+
+.list {
+
```
Comment (luciavelasco) [inline src/components/meetups/MeetupList.module.css]: Remove this empty line

File: src/components/meetups/MeetupList.module.css [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    grid-template-columns: auto auto auto auto;
+    gap: 5px;
+    list-style: none;
+    margin: 0;
+    padding: 0;
+  }
+
+  @media only screen and (max-width: 600px) {
+    .list {
+
```
Comment (luciavelasco) [inline src/components/meetups/MeetupList.module.css]: remove empty lines in css file

File: src/components/meetups/MeetupList.module.css [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+      display: grid;
+      grid-template-columns: auto auto auto;
+
```
Comment (luciavelasco) [inline src/components/meetups/MeetupList.module.css]: remove this empty line

File: src/components/meetups/Meetupitem.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
+import Card from '../ui/Card';
+import classes from './MeetupItem.module.css';
+
+function MeetupItem(props){
+    return (
+    <div className={classes.display}>
```
Comment (luciavelasco) [inline src/components/meetups/Meetupitem.js]: How does this work if you want to add multiple classes to one element?

File: src/components/meetups/Meetupitem.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    <li className={classes.item}>
+        <Card>
+        <div className={classes.image}>
```
Comment (luciavelasco) [inline src/components/meetups/Meetupitem.js]: children of a component should be indented, please fix it to be more like:

```jsx
<Card>
    <div></div>
</Card>
```

not 

```jsx
<Card>
<div></div>
</Card>
```

File: src/components/meetups/Meetupitem.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        </div>
+        <div className={classes.content}>
+            <h3>{props.movie}</h3>
+            <p>{props.genre}</p>
```
Comment (luciavelasco) [inline src/components/meetups/Meetupitem.js]: Each genre should be listed separately

File: src/components/meetups/Meetupitem.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+            <p>{props.year}</p>
+            <p>{props.price}</p>
+            <p>Stock: {props.stock}</p>
+        </div>
+        <div className={classes.actions}>
+            <button>Add to Cart</button>
```
Comment (luciavelasco) [inline src/components/meetups/Meetupitem.js]: Remove this button, as there is no cart until Task 2

File: src/components/ui/Card.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+import classes from './Card.module.css';
+
+function Card(props) {
```
Comment (luciavelasco) [inline src/components/ui/Card.js]: I'm not sure if this needs to be its own element, however I like the use of `children` and you don't need to change this (just something to bear in mind next time).

File: src/index.css [hunk 17]
```diff
@@ -0,0 +1,74 @@
+* {
+  box-sizing: border-box;
+}
+
+body {
+
```
Comment (luciavelasco) [inline src/index.css:6]: empty line, pls remove

File: src/index.css [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+  font-family: sans-serif;
+  margin: 3rem;
+  background-color: #dfdfdf;
+}
+
+h1, h2 {
+  color: #333333;
+}
+
+.btn {
```
Comment (luciavelasco) [inline src/index.css:16]: I prefer using long/full words for class names, i.e. `button` not `btn`.
Optional

File: src/index.css [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+  border: 1px solid #800040;
+  margin: 0 1rem;
+}
+
+.btn:hover {
+  background-color: #9c1458;
+  border-color: #9c1458;
+}
+
+.btn--alt {
```
Comment (luciavelasco) [inline src/index.css]: Interesting double-hyphen convention:
`btn--alt`
`btn-alt`
Make sure the team is on the same page about this.

File: src/index.css [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.2);
+  padding: 1rem;
+  width: 20rem;
+}
+
+.actions {
+  text-align: right;
+}
+
+.modal {
```
Comment (luciavelasco) [inline src/index.css:45]: Is this being used anywhere? If not, please remove this class.

Comment (luciavelasco) [reply src/index.css:45]: If you aren't using it now, but are GOING to use it, you can keep it.

File: src/index.css [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+  padding: 1rem;
+  text-align: center;
+  width: 30rem;
+  z-index: 10;
+  position: fixed;
+  top: 20vh;
+  left: calc(50% - 15rem);
+}
+
+.backdrop {
```
Comment (luciavelasco) [inline src/index.css:58]: same as above, is this being used?

File: src/pages/AllMeetups.js [hunk 22]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+import MeetupList from '../components/meetups/MeetupList';
+import classes from './AllMeetup.css';
+
+      
+function AllMeetupsPage() {
+  const [loadedMeetups, setLoaderMeetups] = useState([]);
+
+
+        fetch ("DVDdata.json")
```
Comment (luciavelasco) [inline src/pages/AllMeetups.js]: Fix the spacing:
`fetch("DVDdata.json")`
not
`fetch ("DVDdata.json")`

File: src/pages/AllMeetups.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+import MeetupList from '../components/meetups/MeetupList';
+import classes from './AllMeetup.css';
+
+      
+function AllMeetupsPage() {
+  const [loadedMeetups, setLoaderMeetups] = useState([]);
```
Comment (luciavelasco) [inline src/pages/AllMeetups.js]: either rename this to dvds, or have a loading state. Optional

**Review by luciavelasco — CHANGES_REQUESTED** [review]
Please rename the word Meetups to Dvds everywhere (use Cmd+Shift+R for easy renaming).

And please use better commit messages, for example:
"Create React App and copy over template"
"Add DVD data to page"
"Fix bug"

**Comment by JoeLam930** [comment]
Renamed "meetupItem" to "DVDsItem"
Renamed "meetuplist" to "DVDslist"
Renamed "Allmeetup" to "AllDVDs"
Renamed "meetups" folder to "DVDsItemsandlists"

**Comment by kely-rufino** [comment]
The PR it's not finished, we need review.

File: package.json [hunk 24]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.16.1",
     "@testing-library/react": "^12.1.2",
     "@testing-library/user-event": "^13.5.0",
+    "create-react-app": "^5.0.0",
```
Comment (luciavelasco) [inline package.json]: I don't think you need this

File: src/components/DVDsItemsandlists/DVDsItem.js [hunk 25]
```diff
@@ -0,0 +1,25 @@
...
+function DVDsItem(props){
+    return (
+    <div>
+    <ul>
+    <li>        
+        <div className={classes.image}>
+        </div>
+        <div className={classes.content}>
+            <h3>{props.movie}</h3>
+            <p>{props.genre}</p>
```
Comment (luciavelasco) [inline src/components/DVDsItemsandlists/DVDsItem.js]: The genres in the `genre` array need to be listed separately e.g.
`props.genre.map(genre => <p>genre</p>)` or `<ul>props.genre.map(genre => <li>genre</li>)</ul>`

File: src/components/DVDsItemsandlists/DVDsItem.js [hunk 26]
```diff
@@ -0,0 +1,25 @@
+import classes from './DVDsItem.css';
+
+function DVDsItem(props){
+    return (
+    <div>
+    <ul>
+    <li>
```
Comment (luciavelasco) [inline src/components/DVDsItemsandlists/DVDsItem.js]: This shouldn't be a list item, and it should not be in an unordered list, as it is only a single item

File: src/components/DVDsItemsandlists/DVDsItem.css [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+    list-style-type: none;
+    display: flex;
+    flex-wrap : wrap;
+    gap: 5px;
+    justify-content: space-around;
+    margin: 0;
+    padding: 0;
+  }
+
+  li {
```
Comment (luciavelasco) [inline src/components/DVDsItemsandlists/DVDsItem.css]: you probably shouldn't have changed this, i think there was a mixup about the genre vs the whole film

**Review by luciavelasco — CHANGES_REQUESTED** [review]
I think three more comments - nearly there, it looks good!!

File: src/components/DVDsItemsandlists/DVDsItem.css (hunk 27 above)
Comment (luciavelasco) [reply src/components/DVDsItemsandlists/DVDsItem.css]: I have no idea how to fix this. When I change it to a class name it doesn't work.

Comment (luciavelasco) [reply src/components/DVDsItemsandlists/DVDsItem.css]: Fixed it, the ul was being used in a different file without being imported directly.

**Review by luciavelasco — APPROVED** [review]
This looks resolved now, I'm happy for this to be merged with main 🎉

**Pull request merged**